src/SubtitlesParser/index.js

```diff
@@ -0,0 +1,199 @@
+    this._lastIndex = 0
+    this._previousCueTimeIndex = 0
+    this._previousCue = ''
+  }
+
+  // get current subtitles
+  // @ currentTime: currentTime in seconds
+  // @return: subtitle text at that currentTime
+  static getSubtitleByTimeIndex(currentTime) {
+    if (typeof currentTime !== 'number' || currentTime === Infinity) {
+      throw new Error('You should pass "currentTime" as a number')
+    }
+
+    if (!Array.isArray(this._captions) || this._captions.length <= 0) {
+      throw new Error('No subtitles available')
+    }
+
+    if (Math.abs(this._previousCueTimeIndex - currentTime) < 0.5) {
+      return this._previousCue
+    }
+
+    if (this._lastIndex > this._captions.length - 1 || !this._lastIndex) {
+      this._lastIndex = 0
+    }
+    const activeIndex = this.getActiveIndex(currentTime) // find active cue from the captions stored
+    this._previousCueTimeIndex = currentTime
+    if (activeIndex !== -1 && activeIndex <= this._captions.length - 1) {
+      this._previousCue = this._captions[activeIndex].payload
+      return this._previousCue
+    } else if (activeIndex === -1) {
+      this._previousCue = ''
+      return ''
+    }
+  }
+
+  static getActiveIndex(currentTime) {
+    let _activeIndex = this._captions
+      .slice(this._lastIndex)
+      .findIndex(cue => currentTime >= cue.start && currentTime < cue.end)
+    if (_activeIndex !== -1) {
+      return _activeIndex + this._lastIndex
+    } else {
+      return this._captions
+        .slice(0, this._lastIndex)
+        .findIndex(cue => currentTime >= cue.start && currentTime < cue.end)
+    }
+  }
+
+  // parses subtitle file and returns list of time, text objects
+  static parseSubtitles(plainSub, parseOptions = {}) {
+    if (parseOptions && 'removeSubtitleTextStyles' in parseOptions) {
+      this._removeSubtitleTextStyles = parseOptions.removeSubtitleTextStyles
+    }
+
+    let linesArray = plainSub
+      .trim()
+      .replace('\r\n', '\n')
+      .split(/[\r\n]/)
+      .map(line => {
+        return line.trim()
+      })
+    let cues = []
+    let start = null
+    let end = null
+    let payload = ''
+    let lines = linesArray.filter(item => item !== '' && isNaN(item))
+    console.log('lines:', lines)
+    for (let i = 0; i < lines.length; i++) {
+      if (lines[i].indexOf('-->') >= 0) {
+        let splitted = lines[i].split(/[ \t]+-->[ \t]+/)
+
+        start = SubtitlesParser.parseTimeStamp(splitted[0])
+        end = SubtitlesParser.parseTimeStamp(splitted[1])
+      } else if (lines[i] !== '') {
+        if (start && end) {
+          if (i + 1 < lines.length && lines[i + 1].indexOf('-->') >= 0) {
+            let subPayload = payload ? payload + ' ' + lines[i] : lines[i]
+            let cue = {
+              start,
+              end,
+              payload: subPayload
+                ? this._removeSubtitleTextStyles
+                  ? subPayload.replace(/<(.*?)>/g, '') // Remove <v- >, etc tags in subtitle text
```

Comment:
I would suggest to split this method up into smaller chunks. The amount of `if` and `else ifs` makes it heard to read. I would also like to see some comments for the use cases of those if/else statements.